### PR TITLE
Create .editorconfig

### PR DESCRIPTION
**Pull request by grst**
This should handle issues like https://github.com/nf-core/modules/pull/30#discussion_r395291647 in all supported editors.

### CODE REVIEW

**Comment by apeltzer** [comment]
Should be also something we need to discuss in nf-core/core for tools for example. @nf-core/core

**Comment by ewels** [comment]
Is it possible to have this file anywhere else? It's a personal pet-peeve of mine when the root directory of repos get polluted with loads of dotfiles (which aren't hidden in the GitHub web interface).

**Comment by grst** [comment]
It will affect the directory it is placed in + all subdirectories. So I guess it's common practice to have it in the repo root.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+max_line_length = 88
```
Comment (ewels) [inline .editorconfig:8]:
```suggestion
```

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_size = 4
+indent_style = space
```
Comment (ewels) [inline .editorconfig:9]:
```suggestion
indent_style = space

[*.{yml,yaml}]
indent_size = 2
```
```

**Comment by ewels** [comment]
I wonder if we can do some of this using `.gitattributes` too? eg. https://github.com/twbs/bootstrap-npm-starter/blob/main/.gitattributes

**Comment by grst** [comment]
I would argue it's best to have both. 
If @ewels can get over it :see_no_evil:

**Comment by ewels** [comment]
I’m over it :) Keen to do as much as possible on this one..

**Pull request merged**